### PR TITLE
Add a solution to initialalg_for_bifunctor_functor

### PR DESCRIPTION
**Pull request by arnoudvanderleer**

### CODE REVIEW

**Comment by arnoudvanderleer** [comment]
I am not sure whether there is a shorter solution. It was one of the most challenging exercises yet. Thereby, also one of the most interesting ones, but I think it is a bit too long (and complicated) to ask on the exam

File: tex/solutions.tex [hunk 1]
```diff
@@ -583,6 +581,64 @@ \section{Solutions}
...
+
+\begin{solution}[\cref{exer:initialalg_for_bifunctor_functor}]\label{sol:initialalg_for_bifunctor_functor}
+	Let $F:\CC\to\CC\to\CC$ be a bifunctor such that for any object $A\in\CC$, the initial algebra $ (\Initalg{F_A}, \In_A) $ for the functor induced by 
+	\[ F_A : \CC\to\CC : X \mapsto F(A,X), \]
+	exists.
+	
+	We define a functor $ G: \CC \to \CC $ by $ G(A) = \Initalg{F_A} $. For $ A, B \in \Ob\CC $, we have an initial $ F_A $-algebra $ (\Initalg{F_A}, \In_A: F_A(\Initalg{F_A}) \to \Initalg{F_A}) $ and an initial $ F_B $-algebra $ (\Initalg{F_B}, \In_B: F_B(\Initalg{F_B}) \to \Initalg{F_B}) $. Now, for all $ f \in \CC(A, B) $, we have by the bifunctoriality of $ F $, a morphism
+	\[ F(\_, \Initalg{F_B})(f): F_A(\Initalg{F_B}) = F(A, \Initalg{F_B}) \to F(B, \Initalg{F_B}) = F_B(\Initalg{F_B}). \]
+	Then $ (\Initalg{F_B}, \In_B \circ F(\_, \Initalg{F_B})(f): F_A(\Initalg{F_B}) \to \Initalg{F_B}) $ is an $ F_A $-algebra, so by the initiality of $ \Initalg{F_A} $, we have a unique morphism $ f^\prime: \Initalg{F_A} \to \Initalg{F_B} $ such that 
+	\[ \In_B \circ F(\_, \Initalg{F_B})(f) \circ F_A(f^\prime) = f^\prime \circ \In_A \]
```
Comment (ghost) [inline tex/solutions.tex:603]: A (commuting) diagram for this equation would be nice.

File: tex/solutions.tex [hunk 2]
```diff
@@ -583,6 +581,65 @@ \section{Solutions}
...
+	\[ \In_C \circ F(\_, \Initalg{F_C})(g \circ f) \circ F_A(G(g \circ f)) = G(g \circ f) \circ \In_A. \]
+	Note that $ F $ is a bifunctor, and therefore, the following diagram commutes:
+	\begin{center}
+		\begin{tikzcd}
+			F(A, \Initalg{F_B}) \arrow[r, "{F(\_, \Initalg{F_B})(f)}"] \arrow[d, "{F(A, \_)(G(g))}"'] & F(B, \Initalg{F_B}) \arrow[d, "{F(B, \_)(G(g))}"]\\
+			F(A, \Initalg{F_C}) \arrow[r, "{F(\_, \Initalg{F_C})(f)}"] & F(B, \Initalg{F_C})
+		\end{tikzcd}
+	\end{center}
+	Then we have the following equality:
+	\begin{align*}
```
Comment (ghost) [inline tex/solutions.tex:633]: Please comment why each argument holds (so maybe at labels to the equations above). You also use associativity here implicit, can you make this explicit?

**Comment by benediktahrens** [comment]
@arnoudvanderleer : thanks a lot for preparing this PR!
@Kfwullaert : thanks for the thorough proof-reading!

**Pull request merged**